proto/spire/api/server/localauthority/v1/localauthority.proto

```diff
@@ -0,0 +1,179 @@
+
+        // TaintX509Authority marks the previously active X.509 authority as
+        // being tainted. SPIRE Agents observing an authority to be tainted
+        // will perform proactive rotations of any key material related to
+        // the tainted authority. The result of this action will be observed
+        // cluster-wide.
+        // It can receive the public key of an old X.509 authority.
+        //
+        // If a previously active X.509 authority does not exist (e.g. if one
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc TaintX509Authority(TaintX509AuthorityRequest) returns (TaintX509AuthorityResponse);
+
+        // RevokeX509Authority revokes the previously active X.509 authority by
+        // removing it from the bundle and propagating this update throughout
+        // the cluster.
+        // It can receive the public key of an old X.509 authority.
+        //
+        // If a previously active X.509 authority does not exist (e.g. if one
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc RevokeX509Authority(RevokeX509AuthorityRequest) returns (RevokeX509AuthorityResponse);
+}
+
+message GetJWTAuthorityStateRequest {}
+
+message GetJWTAuthorityStateResponse {
+        repeated AuthorityState states = 1;
+}
+
+message PrepareJWTAuthorityRequest {}
+
+message PrepareJWTAuthorityResponse {
+        AuthorityState prepared_authority = 1;
+}
+
+message ActivateJWTAuthorityRequest {}
+
+message ActivateJWTAuthorityResponse {
+        AuthorityState activated_authority = 1;
+}
+
+message TaintJWTAuthorityRequest {
+        // The key ID of the local authority to taint
+        string key_id = 1;
+}
+
+message TaintJWTAuthorityResponse {
+        AuthorityState tainted_authority = 1;
+}
+
+message RevokeJWTAuthorityRequest {
+        // The key ID of the local authority to revoke
+        string key_id = 1;
+}
+
+message RevokeJWTAuthorityResponse {
+        AuthorityState revoked_authority = 1;
+}
+
+message GetX509AuthorityStateRequest {}
+
+message GetX509AuthorityStateResponse {
+        repeated AuthorityState states = 1;
+}
+
+message PrepareX509AuthorityRequest {}
+
+message PrepareX509AuthorityResponse {
+        AuthorityState prepared_authority = 1;
+}
+
+message ActivateX509AuthorityRequest {}
+
+message ActivateX509AuthorityResponse {
+        AuthorityState activated_authority = 1;
+}
+
+message TaintX509AuthorityRequest {
+        // The public key of the local authority to taint
+        bytes public_key  = 1;
+}
+
+message TaintX509AuthorityResponse {
+        AuthorityState tainted_authority = 1;
+}
+
+message RevokeX509AuthorityRequest {
+        // The public key of the local authority to revoke
+        bytes public_key  = 1;
+}
+
+message RevokeX509AuthorityResponse {
+        AuthorityState revoked_authority = 1;
+}
+
+message AuthorityState {
```

Comment:
I think we need key ID as part of this message. An operator will first want to list the authorities, and then will want to taint one, and to taint you need the key ID.

Reply:
yeah, I was thinking on adding that based is what we decide about taint and revoke, if unify using public key on both or use key_id for one and public key for another one,
based in what we choose here is what we must return